src/zeit/cms/testing.py

```diff
@@ -437,7 +438,11 @@ def add_by_autocomplete(self, text, widget):
         s.waitForElementPresent(autocomplete_item)
         s.waitForVisible(autocomplete_item)
         s.click(autocomplete_item)
-        s.waitForNotVisible(autocomplete_item)
+        try:
+            s.waitForNotVisible(autocomplete_item)
+        except NoSuchElementException:
+            # in some cases with keywords the menu item is already gone
+            pass
```

Comment:
Hm. Dazu hab ich mehrere Anmerkungen:

1. Wenn du wie oben ein try/except um eine Assertion machst, kannst du sie ebenso gut weglassen, da man nicht sicher sein kann, ob der Zustand getestet wird. Das ist aber nicht hilfreich. In einem solchen Fall muss man das darunter liegende Problem suchen und stattdessen dort ansetzen.
2. Ich nehme an, dass der Fehler bei dir in `zeit.content.article.edit.browser.tests.test_metadata.AuthorLocationTest.test_entering_location_via_autocomplete` aufgetreten ist. Dort bleiben die vorgeschlagenen Wörter aber tatsächlich nach dem auswählen im DOM und bekommen lediglich `display: none`. Ein `waitForNotVisible` ist also tatsächlich die richtige Aktion, verschwinden sollten sie nie. Wenn das passiert, liegt ein Fehler vor. Das sollte man nicht kaschieren, indem man ihn durch try/except zulässt.
3. Eher zu deiner Information: Alle Assertions mit dem Präfix `waitFor` fangen selbst `NoSuchElementException` ab und schlagen erst fehl, wenn auch nach TIMEOUT Sekunden die Assertion nicht zutrifft. In diesem Fall hilft das natürlich nicht (das Schlagwort sollte im DOM bleiben, verschwindet aber ganz). Aber wenn man beispielsweise ein Objekt verschiebt, kann es kurz im DOM nicht gefunden werden und taucht später an anderer Stelle wieder auf. Dafür ist das sehr praktisch.